gallery_dl/extractor/postmill.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for Postmill instances"""
+
+import re
+import urllib.parse
+from .common import BaseExtractor, Message
+from .. import text, exception
+
+
+class PostmillExtractor(BaseExtractor):
+    """Base class for Postmill extractors"""
+    basecategory = "postmill"
+    directory_fmt = ("{category}", "{instance}", "{forum}")
+    filename_fmt = "{id}_{title[:220]}.{extension}"
+    archive_fmt = "{filename}"
+
+    def _init(self):
+        self.instance = self.root.partition("://")[2]
+        self.save_link_post_body = self.config("save-link-post-body", False)
+        self._search_canonical_url = re.compile(r"/f/([\w\d_]+)/(\d+)/").search
+        self._search_image_tag = re.compile(
+            r'<a href="[^"]+"\n +class="submission__image-link"').search
+
+    def items(self):
+        for post_url in self.post_urls():
+            response = self.request(post_url)
```

Comment:
Maybe wrap this in a try/except to prevent a fatal error when one post fails.

Reply:
[Mastodon doesn't do this](https://github.com/mikf/gallery-dl/blob/01bb75f6cbb108279c50c2a89de5c76ef677429b/gallery_dl/extractor/mastodon.py#L34), so I assume it's fine. And if I were to consume an exception, what would I do with it? How could I log it, while somehow making gallery-dl exit with a non-zero error code?

Reply:
Mastodon doesn't need to do it since it doesn't have to send an HTTP request for each single post.

I would have it log a warning about "failing to retrieve post ..." or something and `continue`.

There is an undocumented and incomplete option to change the exit status when logging a certain message, but there is no "native" way to add that in.

Just leave it like it is if you think it's unnecessary.

Reply:
Yeah, I'll probably do nothing about it